src/factories/createSqlTag.ts

```diff
@@ -91,75 +90,81 @@ export const createSqlTag = <T = QueryResultRowType>() => {
       }
     }
 
-    const query = deepFreeze({
+    const query: SqlTokenType = {
       sql: rawSql,
       type: SqlToken,
       values: parameterValues,
+    };
+
+    Object.defineProperty(query, 'sql', {
+      configurable: false,
+      enumerable: true,
+      writable: false,
     });
 
     return query;
   };
 
   sql.array = (
     values: ReadonlyArray<PrimitiveValueExpressionType>,
-    memberType: string | SqlTokenType,
+    memberType: TypeNameIdentifierType | string | SqlTokenType,
```

Comment:
Adding `TypeNameIdentifierType` here should allow auto-complete (intellisense) to supply possible options